lib/promostandards/primary_image_extractor.rb

```diff
@@ -1,6 +1,10 @@
 module PromoStandards
   class PrimaryImageExtractor
-    PRIMARY_IMAGE_PRECEDENCE = ['1006', ['1007', '1001', '2001'], ['1007', '1001'], '1007', ['1001', '2001'], '1001', '1003']
+    PRIMARY_IMAGE_PRECEDENCE = [['1001','2001','1006'], ['1001','2000','1006'], ['1001','2001','1007'],
+                                ['1001','2000','1007'], ['1001','2001','1014'], ['1001','2000','1014'],
+                                ['2001','1006'], ['2000','1006'], ['2001','1007'], ['2000','1007'],
+                                ['2001','1014'], ['2000','1014'], ['1001','2001'], ['1001','2000'],
+                                ['1003', '2001'], '1006', '2001', '2000']
```

Comment:
I was thinking, if we change those single class type ids into arrays, we could avoid checking if it's an array or not here -  https://github.com/promohunt/promostandards-client/pull/11/files#diff-fc54a462395eb053a2e4f8c4b49ff5cdR26

Reply:
I think the intention with single ids is to find inclusion as compared to array? I am not sure though. @manishrc ?

Reply:
He's my understanding:

Say an image has the codes: [1, 2, 3]
I thought that any fo the following will match the above record: [1, 2, 3], [1, 2], 1

Say a Product has the following images: A: [1, 2, 3], B: [1, 2], C: [1, 4], D: [1], E: [5]
Rule [1, 2, 3] should match A
Rule [1, 2] should match A, B
Rule 1 should match images A, B, C, D


Since we specific the more specific rules first, it'll find the image that has a better match first. Once found, it'll abort finding. If not found, it'll continue to iterate.

Reply:
@manishrc In this case, given the rule precedence is [1,2,3], [1,2], 1, we would want to pick A, right?

@amit-choudhary My comment was mostly from a math stand point and was isolated to that little snippet:
`(image_precendence_number - class_type_ids).empty?` checks if `class_type_ids` includes all of `image_precendence_number`

`class_type_ids.include?(image_precendence_number)` also checks if `class_type_ids` includes all of `image_precendence_number` except in this case `image_precendence_number` is just a single entity.

Reply:
Got it. Thanks :)